cscs-checks/system/io/ior_check.py

```diff
@@ -76,86 +87,88 @@ def __init__(self, base_dir):
             )
             data.setdefault('dummy', {})  # entry for unknown systems
 
+    @run_after('init')
+    def set_performance_reference(self):
+        # Converting the references from each fs to per system.
+        self.reference = {
+            '*': self.fs[self.base_dir]['reference']
+        }
+
+    @run_after('init')
+    def set_valid_systems(self):
+        self.valid_systems = self.fs[self.base_dir]['valid_systems']
+
         cur_sys = self.current_system.name
-        if cur_sys not in self.fs[base_dir]:
+        if cur_sys not in self.fs[self.base_dir]:
             cur_sys = 'dummy'
 
-        self.valid_systems = self.fs[base_dir]['valid_systems']
-        self.num_tasks = self.fs[base_dir][cur_sys].get('num_tasks', 1)
-        tpn = self.fs[base_dir][cur_sys].get('num_tasks_per_node', 1)
+        vpe = 'valid_prog_environs'
+        penv = self.fs[self.base_dir][cur_sys].get(vpe, ['builtin'])
+        self.valid_prog_environs = penv
+
+        tpn = self.fs[self.base_dir][cur_sys].get('num_tasks_per_node', 1)
+        self.num_tasks = self.fs[self.base_dir][cur_sys].get('num_tasks', 1)
         self.num_tasks_per_node = tpn
 
-        self.ior_block_size = self.fs[base_dir]['ior_block_size']
-        self.ior_access_type = self.fs[base_dir]['ior_access_type']
-        self.executable_opts = ['-B', '-F', '-C ', '-Q 1', '-t 4m', '-D 30',
-                                '-b', self.ior_block_size,
-                                '-a', self.ior_access_type]
         self.sourcesdir = os.path.join(self.current_system.resourcesdir, 'IOR')
-        self.executable = os.path.join('src', 'C', 'IOR')
-        self.build_system = 'Make'
-
-        vpe = 'valid_prog_environs'
-        penv = self.fs[base_dir][cur_sys].get(vpe, ['builtin'])
-        self.valid_prog_environs = penv
 
+    @run_after('init')
+    def set_build_systems(self):
```

Comment:
```suggestion
    @run_before('compile')
    def prepare_build(self):
```

If you prefer to have everything as a post-init hook, I'm fine.